### PR TITLE
AT E2E: fixes to editor and blocks specs.

### PR DESCRIPTION
**Pull request by worldomonation**
Related to #

## Proposed Changes

This PR fixes the permanent failure encountered in the `Blocks: Core` spec by narrowing the selector.

Key changes:
- match only on visible selectors.
- skip the use of block inserter sidebar search field for faster test execution.

## Testing Instructions

Ensure the following build configurations are passing:
  - [ ] Gutenberg E2E (desktop)
  - [ ] Gutenberg E2E (mobile)
  - [ ] Gutenberg E2E AT (desktop)
  - [ ] Gutenberg E2E AT (mobile)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts:124]: The previous selector ended up matching on more than 1 element on the page, but one of the element was always hidden.

By specifying the `:visible` pseudo-CSS class, we are able to limit the match to visible elements.

File: packages/calypso-e2e/src/lib/components/editor-gutenberg-component.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/components/editor-gutenberg-component.ts:149]: These selectors used to have the `editorPane` to narrow the target, but it ended up causing problems here as the DOM structure of the editor canvas has been in flux (block-based theme, etc).

File: packages/calypso-e2e/src/lib/pages/revisions-page.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/revisions-page.ts]: This is my workaround to the complicated `waitForMutation` call - wait until the network activity has ceased.

`networkidle` has its own set of issues such as when pages inadvertently take longer than expected to load. On the other hand, it's simple to understand and follow along. Any timeouts we can address by adding an appropriate timeout.

File: packages/calypso-e2e/src/lib/pages/revisions-page.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/revisions-page.ts:71]: This bakes in the assumption that navigation initiated from WPAdmin should lead to other WPAdmin pages.

File: test/e2e/specs/blocks/blocks__coblocks__blocks.ts
Comment (worldomonation) [inline test/e2e/specs/blocks/blocks__coblocks__blocks.ts:33]: This is just part of the ongoing `DataHelper.createrSuiteTitle` removal, plus additional notes required for the spec to function.

File: test/e2e/specs/blocks/shared/block-smoke-testing.ts
Comment (worldomonation) [inline test/e2e/specs/blocks/shared/block-smoke-testing.ts:57]: The sidebar block inserter is very slow, especially when searching.

As there exists other specs covering the search and add portion, I've decided that speeding up this spec was preferable.

File: packages/calypso-e2e/src/lib/pages/revisions-page.ts
Comment (dpasque) [reply packages/calypso-e2e/src/lib/pages/revisions-page.ts]: **Question/Comment (blocking)**: could you explain this change a little bit more? Was the `waitForMutation` call being flaky?

I could be misunderstanding, but I'm not sure this is a comparable change, and I'm not sure this will get us what we want over time. 🤔 

Previously, the `Promise.all` check, combined with the `waitForMutations` call, made sure that we knew the revisions view had finished changing _after_ we used the slider.

Now, we're just waiting for networkidle here _before_ we use the slider. So interacting with the revision panel right after this is still prone to race conditions!

Also, I'm not sure checking for `networkidle` this long after a document load is going to do anything. I'm pretty sure that Playwright looks back to see if that event ever fired for this URL/document, so this may just immediately carry on?

**Review by dpasque — CHANGES_REQUESTED** [review]
Thanks for making these fixes @worldomonation! 🎉 

Everything looks good to me, except I'm a little confused by the waitForMutations change. Can you explain that a bit more?

File: packages/calypso-e2e/src/lib/pages/revisions-page.ts
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/revisions-page.ts]: My initial understanding was that `waitForMutation` is checking the stability of the page _before_ the revision was loaded. 

With the corrected spec but without removing the `waitForMutation`, the following is seen because it is leaving behind open handles for Jest to clean up.

```
Ran all test suites matching /test\/e2e\/specs\/editor\/editor__revisions.ts/i.
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

Perhaps I can pose a few questions that will help lead to a decision:
- is there a need to "wait" until the revision is ready to interact, as long as the "load revision" button is ready to interact?
- instead of the complex `waitForMutation`, could we stand in a wait for string we know is part of the revision?

Comment (dpasque) [reply packages/calypso-e2e/src/lib/pages/revisions-page.ts]: > With the corrected spec but without removing the waitForMutation, the following is seen because it is leaving behind open handles for Jest to clean up.

Hmm, weird... 🤔 

Here's what my local testing found:
- Running the test without the `waitForMutations` (as in the current change) actually failed! It looks like it hit the race condition the `waitForMutations` was trying to help with -- the revision content didn't change fast enough, so the validations of that new content ended up failing.
- Running with the `waitForMutations` passed and didn't leave any open handles. 

I wonder if maybe I'm missing something else from your local set-up? Here was my command:
`TEST_ON_ATOMIC=true yarn jest specs/editor/editor__revisions.ts`

> is there a need to "wait" until the revision is ready to interact, as long as the "load revision" button is ready to interact?

"Ready to interact" may be confusing wording here... I think the idea was that we don't want to do validations of the _content of the revisions panel_ until we were sure the new revision content loaded.

> Instead of the complex waitForMutation, could we stand in a wait for string we know is part of the revision?

I think that's a totally valid approach! 👍  I personally don't mind the `waitForMutation` function, and I like that it "future-proofs" a little bit, but we can totally design around this by focusing on specific text content and playwright-waiting for that (instead of getting the `innerText` and doing string assertions against that).

I'm fine either way! 😁  I just think the current approach in the PR might introduce some flakiness.

**Review by dpasque — APPROVED** [review]
Also, going to approve to unblock you as I'm a bit more async today! 😄 I think we should get the Atomic revisions script stable again before merge, but I'm open to however you want to do that!

File: packages/calypso-e2e/src/lib/pages/revisions-page.ts
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/revisions-page.ts]: @dpasque I'll revisit or eliminate the `waitForMutation` changes. It was on the table only because Jest reliably (on my machine) threw the open handle warning, but it wasn't strictly necessary to fix the `Editor: Revision` spec.

**Pull request merged**
